src/jit/lsra.cpp

```diff
@@ -5632,12 +5641,12 @@ void LinearScan::allocateRegisters()
                 }
                 else
 #endif // FEATURE_PARTIAL_SIMD_CALLEE_SAVE
-                    if (currentRefPosition->RequiresRegister() || currentRefPosition->AllocateIfProfitable())
+                    if (currentRefPosition->RequiresRegister() || currentRefPosition->RegOptional())
                 {
                     if (allocateReg)
                     {
-                        assignedRegister = allocateBusyReg(currentInterval, currentRefPosition,
-                                                           currentRefPosition->AllocateIfProfitable());
+                        assignedRegister =
+                            allocateBusyReg(currentInterval, currentRefPosition, currentRefPosition->RegOptional());
```

Comment:
Nice rename. I assume that it is intentional that the parameter of `allocateBusyReg` wasn't also renamed, right?

Reply:
No, that wasn't intentional, just missed. It should probably be renamed as well.

Reply:
I see, I thought that you kept the parameter name just in case, if in the future sources of "profitability" other than "reg optional" may appear.

Reply:
That might make sense, though if anything I might expect that parameter to just be eliminated, since the `RefPosition` should always be `regOptional`, if it doesn't require a register.